src/Hintman/Diff.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeOperators         #-}
+
+{- | This modules contains function to fetch and work with git diffs.
+-}
+
+module Hintman.Diff
+       ( fetchGitHubDiff
+       ) where
+
+import Network.HTTP.Client.TLS (newTlsManager)
+import Servant (PlainText)
+import Servant.Client (BaseUrl (..), ClientM, Scheme (Https), client, mkClientEnv, runClientM)
+import Text.Diff.Parse (parseDiff)
+import Text.Diff.Parse.Types (FileDeltas)
+
+
+{- | Endpoint that represents the following @curl@ command:
+
+@
+curl https://patch-diff.githubusercontent.com/raw/kowainik/hintman/pull/61.diff
+@
+-}
+type GetPullRequestDiffApi =
+    "raw"
+    :> Capture "owner" Text
+    :> Capture "repo" Text
+    :> "pull"
+    :> Capture "number" Text
+    :> Get '[PlainText] Text
+
+-- | Client for 'GetPullRequestDiffApi'.
+getPullRequestDiffClient :: Text -> Text -> Text -> ClientM Text
+getPullRequestDiffClient = client (Proxy @GetPullRequestDiffApi)
+
+{- | Return parsed result of the PR diff.
+-}
+fetchGitHubDiff :: Text -> Text -> Int -> IO (Either String FileDeltas)
+fetchGitHubDiff owner repo prNum = do
```

Comment:
Should these be newtypes?

Reply:
Yes, I guess, we should do that for readability purpose in types